### DISCUSSION TITLE
Wanting to understand plugin's $filePathXXXX variable

### DISCUSSION THREAD

**Discussion by mc12345678** [zencart/zencart#5811] [Q&A]
https://github.com/zencart/zencart/blob/0bf8a8f076843612e87e25966f9f855f901dc2cd/admin/includes/application_bootstrap.php#L170-L189
and
https://github.com/zencart/zencart/blob/35dc5cbc67b99851217a12e580c6130f7b67fd63/includes/application_top.php#L213-L230

Why does `$filePathAdmin` resolve the filepath to: `PLUGIN_DIRECTORY . 'classes/admin'` instead of `PLUGIN_DIRECTORY . 'admin/includes/classes/'`? (Note two things about the original code, the appended content doesn't address the class being in the `admin`'s `includes/classes` folder and that the final character does not include the directory divider.)
and why does `$filePathCatalog` resolve the filepath to: `PLUGIN_DIRECTORY . 'classes/'` instead of `PLUGIN_DIRECTORY . 'catalog/includes/classes/'` (Noting here that the original code *does* resolve to a path that ends with the directory divider.)

Further, why is `$filePathAdmin` even defined in the catalog side? Is consideration being given for catalog side operations to have access to plugin's admin software?

Meanwhile the current direction about plugin directory structure (which doesn't (yet) describe the above filepath) is: https://docs.zen-cart.com/dev/plugins/encapsulated_plugins/directory_structure/

(Update was that in attempting to explain the expected, I still demonstrated an incorrect expectation in the example path.)

**Comment by mc12345678** [comment]
No explanation provided, but looks like it's addressed in #5879. At least that code make sense, even if currently a draft.

**Reply by drbyte** [reply]
Explanation: the plugin architecture code you quoted hasn't been fully-implemented catalog-side (and therefore quite likely has catalog-side bugs if you attempt to use it) due to the zillion ways catalog-side things (mainly templates) get customized. The PR you mentioned aims to fix some of those limitations.

**Comment by lat9** [comment]
Finally (!) corrected via https://github.com/zencart/zencart/pull/6759, in zc210.
